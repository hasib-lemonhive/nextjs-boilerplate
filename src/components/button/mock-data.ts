import {
  IFillButton,
  IFillRedirectButton,
  IOutlineButton,
  IOutlineRedirectButton,
} from './interface';

export const buttonProps: IFillButton = {
  label: 'Button',
  colorScheme: 'primary-bg-text-white',
  hoverColorScheme: 'secondary-bg-text-white',
  size: 'lg',
  clickHandler: () => {},
  'data-testid': 'fill-button',
  shiftIconOnHover: true,
  variant: 'fill',
};

export const redirectButtonProps: IFillRedirectButton = {
  href: 'https://www.google.com/',
  isOpenNewTab: true,
  label: 'Redirect Button',
  colorScheme: 'primary-bg-text-white',
  hoverColorScheme: 'secondary-bg-text-white',
  size: 'lg',
  'data-testid': 'fill-redirect-button',
  variant: 'fill',
};

export const outlineButtonProps: IOutlineButton = {
  label: 'Outline Button',
  colorScheme: 'transparent-bg-text-primary',
  hoverColorScheme: 'transparent-bg-text-secondary',
  size: 'lg',
  clickHandler: () => {},
  'data-testid': 'outline-button',
  variant: 'outline',
};

export const redirectOutlineButtonProps: IOutlineRedirectButton = {
  href: 'https://www.google.com/',
  isOpenNewTab: true,
  label: 'Outline Redirect Button',
  colorScheme: 'transparent-bg-text-primary',
  hoverColorScheme: 'transparent-bg-text-secondary',
  size: 'lg',
  'data-testid': 'outline-redirect-button',
  variant: 'outline',
};
