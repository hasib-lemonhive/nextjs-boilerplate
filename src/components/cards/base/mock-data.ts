import { CardBorderRadiusEnum, CardShadowEnum, IBaseCard } from './interface';

export const baseCardProps: IBaseCard = {
  dataTestId: 'base-card',
  cardBorderRadius: CardBorderRadiusEnum.RadiusOne,
  cardShadow: CardShadowEnum.ShadowOne,
};
