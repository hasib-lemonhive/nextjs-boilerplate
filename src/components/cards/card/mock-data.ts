import { IDog } from './interface';

export const customPortableTextRaw = [
  {
    _key: '99b334a94800',
    _type: 'customBlock',
    children: [
      {
        _key: 'f0688e8be4e50',
        _type: 'span',
        marks: [],
        text: 'About the Breed',
      },
    ],
    markDefs: [],
    style: 'headingOne',
  },
  {
    _key: 'af61f6e66ed5',
    _type: 'customBlock',
    children: [
      {
        _key: 'd77faf4742ac0',
        _type: 'span',
        marks: [],
        text: 'The Alaskan Klee Kai is a small-sized companion dog that is alert, energetic, and curious, yet reserved with unfamiliar people and situations.',
      },
    ],
    markDefs: [],
    style: 'normal',
  },
  {
    _key: '363ef2e8e1ee',
    _type: 'customBlock',
    children: [
      {
        _key: 'd16017cdc8710',
        _type: 'span',
        marks: [],
        text: 'The Alaskan Klee Kai is a small sized companion dog with a smooth, agile, and well-balanced gait, a body that is well proportioned with a level topline and a length slightly longer than height. The wedge-shaped head, erect triangle shaped ears, well furred double coat in three symmetrical contrasting color varieties, and a loosely curled tail reflect its arctic heritage.',
      },
    ],
    markDefs: [],
    style: 'normal',
  },
  {
    _key: '822d0f91ac4f',
    _type: 'customBlock',
    children: [
      {
        _key: '7f10ffe37bd50',
        _type: 'span',
        marks: [],
        text: '',
      },
    ],
    markDefs: [],
    style: 'normal',
  },
];

/**
 * You should only use type assertion (as) in mock data.
 */
export const mockDog = {
  breed: 'Test Breed',
  name: 'Test Name',
  description: customPortableTextRaw,
  image: {
    src: 'https://assets.imgix.net/unsplash/jellyfish.jpg?w=800&h=400&fit=crop&crop=entropy',
    alt: 'Test Alt',
    lqip: 'https://assets.imgix.net/unsplash/jellyfish.jpg?w=800&h=400&fit=crop&crop=entropy&px=16&blur=200&fm=webp',
  },
} as IDog;
