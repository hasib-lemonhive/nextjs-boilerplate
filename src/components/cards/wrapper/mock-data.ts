import {
  CardBorderRadiusEnum,
  CardShadowEnum,
  ICardWrapper,
} from './interface';

export const baseCardProps: ICardWrapper = {
  cardBorderRadius: CardBorderRadiusEnum.RadiusOne,
  cardShadow: CardShadowEnum.ShadowOne,
};
