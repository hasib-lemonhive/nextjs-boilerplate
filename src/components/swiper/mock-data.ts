export const swiperPreviewimages: string[] = [
  'https://res.cloudinary.com/dktw59lc2/image/upload/v1669267968/BrightonSEO/Carousel/image-1_n2g4dh.jpg',
  'https://res.cloudinary.com/dktw59lc2/image/upload/v1669267967/BrightonSEO/Carousel/image-2_yph8q6.jpg',
  'https://res.cloudinary.com/dktw59lc2/image/upload/v1669267968/BrightonSEO/Carousel/image-3_zj0pxz.jpg',
  'https://res.cloudinary.com/dktw59lc2/image/upload/v1669267968/BrightonSEO/Carousel/image-4_pwi0a2.jpg',
  'https://res.cloudinary.com/dktw59lc2/image/upload/v1669267968/BrightonSEO/Carousel/image-5_ptsm14.jpg',
  'https://res.cloudinary.com/dktw59lc2/image/upload/v1669267968/BrightonSEO/Carousel/image-6_pbvbnf.jpg',
];

export const swiperCommonOptions = {
  isCenter: false,
  isLoop: false,
  autoPlay: false,
  speed: 800,
};
