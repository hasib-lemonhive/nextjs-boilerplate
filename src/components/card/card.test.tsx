import { render } from '@testing-library/react';
import Card from '.';
import { mockDog } from './mock-data';

it('renders Buy Ticket Card correctly', () => {
  const { getByRole } = render(
    <Card link="https://www.google.com" dog={mockDog} />
  );
  const element = getByRole('base-card');
  expect(element).toBeInTheDocument();
  expect(element).toMatchSnapshot();
});
