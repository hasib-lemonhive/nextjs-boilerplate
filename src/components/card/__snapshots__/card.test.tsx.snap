// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Buy Ticket Card correctly 1`] = `
<div
  class="p-10"
  role="base-card"
>
  <div
    class="card"
  >
    <img
      alt="Test Alt"
      data-nimg="1"
      decoding="async"
      height="400"
      loading="lazy"
      src="http://localhost/_next/image?url=https%3A%2F%2Fassets.imgix.net%2Funsplash%2Fjellyfish.jpg%3Fw%3D800%26h%3D400%26fit%3Dcrop%26crop%3Dentropy&w=1200&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fassets.imgix.net%2Funsplash%2Fjellyfish.jpg%3Fw%3D800%26h%3D400%26fit%3Dcrop%26crop%3Dentropy&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fassets.imgix.net%2Funsplash%2Fjellyfish.jpg%3Fw%3D800%26h%3D400%26fit%3Dcrop%26crop%3Dentropy&w=1200&q=75 2x"
      style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
      width="600"
    />
    <div
      class="px-6 py-4"
    >
      <div
        class="font-bold text-xl mb-2"
      >
        Test Name
      </div>
      <div
        class="text-gray-700 dark:text-slate-200 text-base truncate"
      >
        <p>
          About the Breed
        </p>
        <p>
          The Alaskan Klee Kai is a small-sized companion dog that is alert, energetic, and curious, yet reserved with unfamiliar people and situations.
        </p>
        <p>
          The Alaskan Klee Kai is a small sized companion dog with a smooth, agile, and well-balanced gait, a body that is well proportioned with a level topline and a length slightly longer than height. The wedge-shaped head, erect triangle shaped ears, well furred double coat in three symmetrical contrasting color varieties, and a loosely curled tail reflect its arctic heritage.
        </p>
        <p />
      </div>
    </div>
    <div
      class="px-6 pt-4 pb-2"
    >
      <span
        class="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2 mb-2"
      >
        #tag1
      </span>
      <span
        class="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2 mb-2"
      >
        #tag2
      </span>
      <span
        class="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2 mb-2"
      >
        #tag3
      </span>
    </div>
  </div>
</div>
`;
